src/ripple/nodestore/impl/Shard.cpp

```diff
@@ -556,7 +567,10 @@ Shard::finalize(const bool writeSQLite)
             return fail("missing root TXN node");
         }
 
-        if (!valLedger(ledger, next))
+        if (dsh)
+            dsh->store(nObj);
+
+        if (!valLedger(ledger, next, dsh))
```

Comment:
While you're here, please rename this function `verifyLedger`. Rationale: `valLedger` is a poor name and the term `validate` has a very different meaning in our code; we should avoid confusing. Similarly, please reame the message below: "verification check failed"

Reply:
No problem if the author (@miguelportilla if I am not mistaking) agree with it.

Reply:
`verifyLedger` is okay with me.

Reply:
Done